src/components/NcAppSidebar/NcAppSidebar.vue

```diff
@@ -565,107 +565,111 @@ export default {
+								'app-sidebar-header__desc--without-actions': !$slots['secondary-actions'],
+							}"
+							class="app-sidebar-header__desc">
+							<!-- favourite icon -->
+							<div v-if="canStar || $slots['tertiary-actions']" class="app-sidebar-header__tertiary-actions">
+								<slot name="tertiary-actions">
+									<NcButton v-if="canStar"
+										:aria-label="favoriteTranslated"
+										:pressed="isStarred"
+										class="app-sidebar-header__star"
+										variant="secondary"
+										@click.prevent="toggleStarred">
+										<template #icon>
+											<NcLoadingIcon v-if="starLoading" />
+											<Star v-else-if="isStarred" :size="20" />
+											<StarOutline v-else :size="20" />
+										</template>
+									</NcButton>
 								</slot>
-							</p>
+							</div>
+
+							<!-- name -->
+							<div class="app-sidebar-header__name-container">
+								<div class="app-sidebar-header__mainname-container">
+									<!-- main name -->
+									<h2 v-show="!nameEditable"
+										:id="`app-sidebar-vue-${uid}__header`"
+										ref="header"
+										v-linkify="{text: name, linkify: linkifyName}"
+										:aria-label="title"
+										:title="title"
+										class="app-sidebar-header__mainname"
+										:tabindex="nameEditable ? 0 : -1"
+										@click.self="editName">
+										{{ name }}
+									</h2>
+									<template v-if="nameEditable">
+										<form v-click-outside="() => onSubmitName()"
+											class="app-sidebar-header__mainname-form"
+											@submit.prevent="onSubmitName">
+											<input ref="nameInput"
+												v-focus
+												class="app-sidebar-header__mainname-input"
+												type="text"
+												:placeholder="namePlaceholder"
+												:value="name"
+												@keydown.esc.stop="onDismissEditing"
+												@input="onNameInput">
+											<NcButton :aria-label="changeNameTranslated"
+												type="submit"
+												variant="tertiary-no-background">
+												<template #icon>
+													<ArrowRight :size="20" />
+												</template>
+											</NcButton>
+										</form>
+									</template>
+									<!-- header main menu -->
+									<NcActions v-if="$slots['secondary-actions']"
+										class="app-sidebar-header__menu"
+										:force-menu="forceMenu">
+										<slot name="secondary-actions" />
+									</NcActions>
+								</div>
+								<!-- secondary name -->
+								<p v-if="subname.trim() !== '' || $slots['subname']"
+									:title="subtitle || undefined"
+									class="app-sidebar-header__subname">
+									<!-- @slot Alternative to the `subname` prop can be used for more complex conent. It will be rendered within a `p` tag. -->
+									<slot name="subname">
+										{{ subname }}
+									</slot>
+								</p>
+							</div>
 						</div>
 					</div>
-				</div>
+				</slot>
 
-				<NcButton ref="closeButton"
+				<NcButton v-if="!noClose"
+					ref="closeButton"
```

Comment:
When there is no button
```
TypeError: Cannot read properties of undefined (reading 'focus')
```

Reply:
Caused by `empty` prop, when there is no `$refs.header` rendered